src/app/Shared/Services/Api.service.tsx

```diff
@@ -503,8 +503,7 @@ export class ApiService {
 
   // from file system path functions
   uploadArchivedRecordingToGrafanaFromPath(jvmId: string, recordingName: string): Observable<boolean> {
-    const subdirectoryName = jvmIdToSubdirectoryName(jvmId);
-    return this.sendRequest('beta', `fs/recordings/${subdirectoryName}/${encodeURIComponent(recordingName)}/upload`, {
+    return this.sendRequest('beta', `fs/recordings/${jvmId}/${encodeURIComponent(recordingName)}/upload`, {
```

Comment:
I think the `jvmId` should also be passed through `encodeURIComponent` to be safe. It may be the case currently that the IDs the backend assigns are always URI-safe, but this way we can future-proof against that ever changing.